### PR TITLE
[Term Entry] Java ArrayList: .spliterator() added

### LINKED ISSUE #4103: [Term Entry] Java ArrayList: .spliterator()
### Reason/inspiration (optional)

We want to add the `.spliterator()` method as a new term entry for [ArrayList](https://www.codecademy.com/resources/docs/java/array-list). Provide the entry in a markdown file and place it under the file path `content/java/concepts/array-list/terms/spliterator/spliterator.md`

The entry should have the following:
– An introductory paragraph about what the method does.
– A syntax section with a pseudo code block explaining it.
– An example showing the method in use.

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by antriksh-9**
### Description

I was assigned this [issue](https://github.com/Codecademy/docs/issues/4103).
Added new entry for Java terms: ArrayList .spliterator()
Closes #4103.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by Sriparno08 — COMMENTED** [review]
Hey @antriksh-9, thanks for contributing to Codecademy!

Before we move on to our first review process, you'll need to fix some instances that are identified as plagiarized content.

These instances are listed after this comment. Please review them and rewrite them in a way that is unique.

**Comment by antriksh-9** [comment]
Hey @Sriparno08 I have made the necessary changes.

**Comment by Sriparno08** [comment]
Good job, @antriksh-9 👍

**Review by Sriparno08 — COMMENTED** [review]
Hey @antriksh-9, here are the modifications that you need to make in your article:

**Comment by antriksh-9** [comment]
Hey @Sriparno08 I have made the necessary changes.

**Review by Sriparno08 — APPROVED** [review]
Well done, @antriksh-9!

This looks for a second review 👍

File: content/java/concepts/array-list/terms/spliterator/spliterator.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+```pseudo
+public Spliterator<T> split = list.spliterator()
+```
+
+Here, `T` represents the type of elements in the `list` [collection](https://www.codecademy.com/resources/docs/java/collection). For example, if the collection is of type `List<String>`, i.e., it contains [`String`](https://www.codecademy.com/resources/docs/java/strings) elements, then `T` would be `String`.
+
+## Examples
+
+Let's take a look at some examples to understand the `.spliterator()` method.
```
Comment (SSwiniarski) [inline content/java/concepts/array-list/terms/spliterator/spliterator.md:36]:
```suggestion
```

This isn't necessary and we avoid first and second pronouns in docs (direct address to the reader.)

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@antriksh-9 I did a second review and only had one comment.

**Comment by antriksh-9** [comment]
Hey @SSwiniarski I have made the necessary change.

File: content/java/concepts/array-list/terms/spliterator/spliterator.md (hunk 1 above)
Comment (Sriparno08) [reply content/java/concepts/array-list/terms/spliterator/spliterator.md]: Yeah, but @SSwiniarski is right here 👍

**Pull request merged**